### PR TITLE
gh #190 dsvideo port l3 C and Python testcase Implementation 

### PR DESCRIPTION
**Pull request by hari22yuva**
dsVideoPort L3 C and Python testcase Implementation

### CODE REVIEW

File: host/tests/dsClasses/dsVideoPort.py [hunk 1]
```diff
@@ -0,0 +1,1465 @@
...
+
+    def getResolutions(self):
+        """
+        Returns the supported Resolutions on device.
+
+        Args:
+            None.
+
+        Returns:
+            returns the supported resolutions
```
Comment (Ulrond) [inline host/tests/dsClasses/dsVideoPort.py:1232]: you need to be specific about the data that's returned and what it looks like.

File: host/tests/dsClasses/dsVideoPort.py [hunk 2]
```diff
@@ -0,0 +1,1465 @@
...
+
+    def getAspectRatio(self):
+        """
+        Returns the supported aspectRatio on device.
+
+        Args:
+            None.
+
+        Returns:
+            returns the supported aspectRatio
```
Comment (Ulrond) [inline host/tests/dsClasses/dsVideoPort.py:1276]: you need to be specific about what format the returned data is in, and give an example.

**Review by Ulrond — DISMISSED** [review]
# dsXXX Review Points


## Module Class `dsXxx`

- needs to be self contained, and therefore when you run class.init() it will required to install and copy to the target any requirements to meet running.
- the class configuration must include all information required to copy files ( as required ), to the target and deploy
- This is because it will be called from L4 class functions and therefore L4 will not know how to configure or control each individual module. 
`dsClassxxx`

Therefore all it's configuration needs to be in `dsxxx_test_Suite.yml`, which is how the test works, and what the menu's are in the test itself, which asserts need to be downloaded to fulfil the requirements for the test. You can assume that the binaries are local, and relative to this file, because it sits in a fixed location within the testing suite... And the binaries are fixed because of this version of the C Code which is within this codebse.

- Build the test, you get a yaml file output, which contains both the menu's for the current version of the suite, and is filled out with the binaries that are required to be copied to the target.
- Copy the binaries based on the Class Configuration to the target.

## Testing Suites

General rules:

1. no function can repeat implenentation ( zero repeats ) , if you write the same function twice, then it's a generic function that moves to the module specific `helperClass`.
	- Functions that fall into this category
		- same initialisation across all tests in  `__init__()`, which means it's repeating, the same as all these functions.
		- `testDownloadAssets()`
		- `testCleanAssets()`
		- `testRunPrerequisites()` -> Should be a in helper `testController` `testPrepareFunction()`
		- `testCleanAssets()` -> should be called from `testPrepareFunction()`
		- `testExceptionCleanUp` -> Should be used and should call `testCleanAssets()`
2. Tests are platform independant, and must not contain any platform specific assumptions
3. Platform specific configuration needs to reside in `deviceConfig.yaml` where each `cpe` can define anything platform specific as a variable. Therefore the same variable can contain different information across the platforms.
4. Tests must be able to loop without re-downloading assets, framework  `testPrepareFunction()` & `testExitFunction()` & `testExceptionCleanUp()` functions should be used in these case where assets are to be added / remove from the platform.
5. OpenLogs or any other files once, that's what init(), & deInit() are for, whilst the testing suite is open, the logs for callbacks, etc. should remain open. (callbacks should not open and close the logs, they should only write to the currently active log). Filenames for the logs etc. are specific and can be statically deinfed in the `dsXXXClass()` but you need functions to return the currently active filename... `dsXXXClass.getCallbackFiles()` etc.
6. Callback logging needs to be added to the `utHelperClass` so that its common format for all tests across all modules. (Filenames for the logs should be passed from the caller)
7. All functions and what is returned from `dsxxxClass` must be fully documented, on what type is returned, list etc, and what format information is returned. (Other callers outside of the testing suite needs to know how to call the class)

## UTHelper classes

- playerClass
	- Is self contained, and must not have `testSpecific` configuration
	- It must work on all platforms, but can be configured by passing the `cpe` playback type... Which can be defined in the `deviceConfig.yaml`

**Review by bhanucbp — APPROVED** [review]
ship it

**Review by srinivasgtl — APPROVED** [review]
approved.  Need to address the comments in the next phase.

**Pull request merged**